### PR TITLE
Calico ipv6

### LINKED ISSUE #240: The IPv6 address is not assigned on Debian
The Debian box we're using doesn't use NetworkManager to manage connections, so we've to find another way to add an IPv6 address, or let NetworkManager manage the interface.

Important for #192

### PR DESCRIPTION
**Pull request by StefanoSalsano**
Fixes #240

### CODE REVIEW

**Comment by ferrarimarco** [comment]
Hi @StefanoSalsano 

Is this something you're still working on? Thanks!

**Comment by StefanoSalsano** [comment]
yes sure !! it's top priority... so it will come after the top-top priority issues on which we're working now :-)

**Comment by ferrarimarco** [comment]
Nice! P -1 :)

**Comment by ferrarimarco** [comment]
@StefanoSalsano do we still need this? Do you remember how much work was needed to complete this feature? Thanks!

**Comment by StefanoSalsano** [comment]
> 
> 
> @StefanoSalsano do we still need this? Do you remember how much work was needed to complete this feature? Thanks!

@ferrarimarco yes we still need this strongly :-) I was really close to reach the goal so I think that we do not need much work, because this issue is only concerned with the basic support of IPv6 in Calico... we have to merge the latest changes from master into this branch and proceed... @zvfvrv will have to work on this in the next weeks

**Comment by ferrarimarco** [comment]
Nice! Leaving this out of a specific milestone for now. We can update planning after we resume work on this PR. Thanks!

**Comment by zvfvrv** [comment]
will fix #240

File: ansible/roles/kubernetes/tasks/setup-Debian.yml [hunk 1]
```diff
@@ -24,3 +24,5 @@
       - python-selinux
       - selinux-policy-default
       - snapd
+      - ipvsadm
+      - vim
```
Comment (ferrarimarco) [inline ansible/roles/kubernetes/tasks/setup-Debian.yml:28]: Why do we need vim?

File: defaults.yaml [hunk 2]
```diff
@@ -65,8 +65,8 @@ net:
   master_ipv6_part: "c40a::"
   # base IPv6 address suffix for the minion nodes.
   # must be a string with 4 hex digits followed by "::"
-  # minion_ipv6_part: "c41e:"
-  minion_ipv6_part: "0001::"
+  minion_ipv6_part: "c41e::"
+  #minion_ipv6_part: "0001::"
```
Comment (ferrarimarco) [inline defaults.yaml:69]: Why some of these are commented out?

File: scripts/linux/configure-ipv6-interface.sh [hunk 3]
```diff
@@ -5,14 +5,23 @@ echo "Configuring the IPv6 interface"
...
-nmcli con mod "$UUID" ipv6.address "$IPV6_ADDRESS/$IPV6_NETMASK" ipv6.method manual
 
-echo "Restarting NetworkManager"
-systemctl restart network
+{
+    echo "auto $DEVICE"
+    echo "iface $DEVICE inet6 static"
+    echo "  address $IPV6_ADDRESS"
+    echo "  netmask $IPV6_NETMASK"
+} >>"$NET_CONFIG_PATH"
```
Comment (ferrarimarco) [inline scripts/linux/configure-ipv6-interface.sh:17]: This will append to the file, so if you reload the configuration, it adds this stuff every time?

Comment (ferrarimarco) [reply scripts/linux/configure-ipv6-interface.sh:17]: Why don't we use Ansible for this? So we don't have to care about checking if the content is there before adding.

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Hi @zvfvrv ! Thanks for the PR.

Great to see this moving forward.

You're on the right track. I've just a couple of minor comments on what you provided here.

Can you please add the tests as well? To validate that:

- [x] The interface has the expected IPv6 address assigned
- [x] The `ipvsadm` package is installed (I feel we don't need the `vim` package, so no need to test for that, see my other comment)
- [ ] `/etc/network/interfaces` has the IPv6 content you add in `configure-ipv6-interface.sh`

This should all take minutes because there are already examples where we're doing very similar things in the test suite.

**Comment by ferrarimarco** [comment]
PS: I suggest you run InSpec locally, so you don't have to wait for the whole build to complete.

File: ansible/roles/kubernetes/tasks/setup-Debian.yml (hunk 1 above)
Comment (zvfvrv) [reply ansible/roles/kubernetes/tasks/setup-Debian.yml:28]: i use vim inside the cluster, and each time i destroy it and then rebuild i have to reinstall it :) 
so if you don't mind i would like to keep it as basic tool :)

File: defaults.yaml (hunk 2 above)
Comment (zvfvrv) [reply defaults.yaml:69]: I found them already there! I will remove them

File: ansible/roles/kubernetes/tasks/setup-Debian.yml (hunk 1 above)
Comment (ferrarimarco) [reply ansible/roles/kubernetes/tasks/setup-Debian.yml:28]: /shrug

File: scripts/linux/configure-ipv6-interface.sh (hunk 3 above)
Comment (zvfvrv) [reply scripts/linux/configure-ipv6-interface.sh:17]: are you suggesting me to use a template?

Comment (ferrarimarco) [reply scripts/linux/configure-ipv6-interface.sh:17]: Fur such a small addition, I suppose that `blockinfile` is enough.

File: defaults.yaml (hunk 2 above)
Comment (ferrarimarco) [reply defaults.yaml:69]: Oh, I didn't know. Thanks!

File: scripts/linux/configure-ipv6-interface.sh (hunk 3 above)
Comment (zvfvrv) [reply scripts/linux/configure-ipv6-interface.sh:17]: ok done

**Comment by ferrarimarco** [comment]
Nice work! I'll wait for the results of the build and then merge. Thanks!

**Comment by zvfvrv** [comment]
Great thanks!

**Comment by ferrarimarco** [comment]
Ok the build passed. Let's merge the latest changes from master...

**Pull request merged**